packages/js/src/workouts/tailwind-components/steps/personal-preferences/newsletter-signup.js

```diff
@@ -0,0 +1,154 @@
+import { useCallback, useState, Fragment } from "@wordpress/element";
+import { __, sprintf } from "@wordpress/i18n";
+import { isEmail } from "@wordpress/url";
+import PropTypes from "prop-types";
+
+import { addLinkToString } from "../../../../helpers/stringHelpers";
+import TextInput from "../../base/text-input";
+
+/**
+ * A function to request a sign up to the newsletter.
+ *
+ * @param {string} email The email to signup to the newsletter.
+ *
+ * @returns {Object} The request's response.
+ */
+async function postSignUp( email ) {
+	const response = await fetch( "https://my.yoast.com/api/Mailing-list/subscribe", {
+		method: "POST",
+		mode: "cors",
+		cache: "no-cache",
+		credentials: "same-origin",
+		headers: {
+			"Content-Type": "application/json",
+		},
+		redirect: "follow",
+		referrerPolicy: "no-referrer",
+		body: JSON.stringify(
+			{
+				customerDetails: {
+					firstName: "",
+					email,
+				},
+				list: "Yoast newsletter",
+			}
+		),
+	} );
+	return response.json();
+}
+
+const genericErrorFeedback = __( "Oops! Something went wrong. Check your email address and try again.", "wordpress-seo" );
+const invalidEmailFeedback = __( "That is not a valid email address. Check your email address and try again.", "wordpress-seo" );
+const alreadySubscribedFeedback = __( "That email address has already been subscribed.", "wordpress-seo" );
+const subscribedFeedback = __(
+	"Thanks! Please click the link in the email we just sent you to confirm your newsletter subscription.",
+	"wordpress-seo"
+);
+
+/**
+ * The newsletter signup section.
+ *
+ * @param {string} gdprLink Shortlink to the Yoast privacy policy.
+ *
+ * @returns {WPElement} A newslettersignup element.
+ */
+export function NewsletterSignup( { gdprLink } ) {
+	const [ newsletterEmail, setNewsletterEmail ] = useState( "" );
+	const [ signUpState, setSignUpState ] = useState( "waiting" );
+	const [ emailFeedback, setEmailFeedback ] = useState( "" );
+
+	const onSignUpClick = useCallback(
+		async function() {
+			if ( ! isEmail( newsletterEmail ) ) {
+				setSignUpState( "error" );
+				setEmailFeedback( invalidEmailFeedback );
+				return;
+			}
+			setSignUpState( "loading" );
+			const response = await postSignUp( newsletterEmail );
+			if ( response.error ) {
+				setSignUpState( "error" );
+				setEmailFeedback( genericErrorFeedback );
```

Comment:
```suggestion
				setEmailFeedback( genericErrorFeedback );
				setEmailFeedback( ( prevState ) => [ ...prevState, genericErrorFeedback ] );
```

Reply:
As mentioned in our call, no need to add new feedback to the prevstate, since every submission of this field will lead to one (set of) error(s)